sdk/helper/testcluster/docker/environment.go

```diff
@@ -0,0 +1,963 @@
+	scanner := bufio.NewScanner(bytes.NewReader(p))
+	scanner.Buffer(make([]byte, 64*1024), bufio.MaxScanTokenSize)
+	for scanner.Scan() {
+		l.consumer(scanner.Text())
+	}
+	return len(p), nil
+}
+
+// DockerClusterOptions has options for setting up the docker cluster
+type DockerClusterOptions struct {
+	testcluster.ClusterOptions
+	CAKey       *ecdsa.PrivateKey
+	NetworkName string
+	ImageRepo   string
+	ImageTag    string
+	CloneCA     *DockerCluster
+	VaultBinary string
+}
+
+func ensureLeaderMatches(ctx context.Context, client *api.Client, ready func(response *api.LeaderResponse) error) error {
+	var leader *api.LeaderResponse
+	var err error
+	for ctx.Err() == nil {
+		leader, err = client.Sys().Leader()
+		switch {
+		case err != nil:
+		case leader == nil:
+			err = fmt.Errorf("nil response to leader check")
+		default:
+			err = ready(leader)
+			if err == nil {
+				return nil
+			}
+		}
+		time.Sleep(500 * time.Millisecond)
+	}
+	return fmt.Errorf("error checking leader: %v", err)
+}
+
+const DefaultNumCores = 3
+
+// creates a managed docker container running Vault
+func (dc *DockerCluster) setupDockerCluster(ctx context.Context, opts *DockerClusterOptions) error {
+	if opts.TmpDir != "" {
+		if _, err := os.Stat(opts.TmpDir); os.IsNotExist(err) {
+			if err := os.MkdirAll(opts.TmpDir, 0o700); err != nil {
+				return err
+			}
+		}
+		dc.tmpDir = opts.TmpDir
+	} else {
+		tempDir, err := ioutil.TempDir("", "vault-test-cluster-")
+		if err != nil {
+			return err
+		}
+		dc.tmpDir = tempDir
+	}
+	caDir := filepath.Join(dc.tmpDir, "ca")
+	if err := os.MkdirAll(caDir, 0o755); err != nil {
+		return err
+	}
+
+	var numCores int
+	if opts.NumCores == 0 {
+		numCores = DefaultNumCores
+	} else {
+		numCores = opts.NumCores
+	}
+
+	if opts.CloneCA != nil {
+		dc.CACert = opts.CloneCA.CACert
+		dc.CACertBytes = opts.CloneCA.CACertBytes
+		dc.CACertPEM = opts.CloneCA.CACertPEM
+		dc.CACertPEMFile = opts.CloneCA.CACertPEMFile
+		dc.CAKey = opts.CloneCA.CAKey
+		dc.CAKeyPEM = opts.CloneCA.CAKeyPEM
+		dc.RootCAs = opts.CloneCA.RootCAs
+	} else {
+		if err := dc.setupCA(opts); err != nil {
+			return err
+		}
+	}
+
+	for i := 0; i < numCores; i++ {
+		if err := dc.addNode(ctx, opts); err != nil {
+			return err
+		}
+		if opts.SkipInit {
+			continue
+		}
+		if i == 0 {
+			if err := dc.setupNode0(ctx); err != nil {
+				return nil
+			}
+		} else {
+			if err := dc.joinNode(ctx, i, 0); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (dc *DockerCluster) AddNode(ctx context.Context, opts *DockerClusterOptions) error {
+	leaderIdx, err := testcluster.LeaderNode(ctx, dc)
+	if err != nil {
+		return err
+	}
+	if err := dc.addNode(ctx, opts); err != nil {
+		return err
+	}
+
+	return dc.joinNode(ctx, len(dc.ClusterNodes)-1, leaderIdx)
+}
+
+func (dc *DockerCluster) addNode(ctx context.Context, opts *DockerClusterOptions) error {
+	i := len(dc.ClusterNodes)
+	nodeID := fmt.Sprintf("core-%d", i)
+	node := &dockerClusterNode{
```

Comment:
Ah, sorry in `RunNginxRootTest` we call `..., networkName, ... := buildNginxContainer` which creates a container on an arbitrary network and subsequent calls which create containers which are supposed to be on the same network take that in e.g., `CheckWithClients(t, networkName, ...)` -- to ensure they're created on the same network as the nginx instance. 

Here, since we're exposing this in `sdk` as a generic test dependency, we can't know how the container would get created, so it'd be best to ensure that the subsequent nodes land on the same network (via forcing it to) as the first node.